Keycloak arquillian testsuite not working with the default profile

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

testsuite

### Describe the bug

The testsuite does not work by default. It is executed on embedded undertow by default, which was broken by https://github.com/keycloak/keycloak/commit/5c901016e7ebdf5c5eebe2aea948a03df172de03#diff-7f2f15dbc5b7f3411286e328eef100d4247bf7cfe0317bde84c187a64f4be085



### Version

nightly (from 2024-12-11)

### Regression

- [X] The issue is a regression

### Expected behavior

Keycloak testsuite working with the default server (embedded undertow)

### Actual behavior

Keycloak testsuite not working with the default server (embedded undertow)

### How to Reproduce?

1) Default tests with embedded undertow are not working. For example this one is failing:

```
cd testsuite/integration-arquillian/tests/base/
mvn clean install -Dtest=LoginTest
```


2) Running embedded undertow does not work:

```
cd testsuite/utils
mvn exec:java -Pkeycloak-server
```


In both cases, there is similar error like:
```
Caused by: java.util.ServiceConfigurationError: org.keycloak.config.ConfigProviderFactory: Provider org.keycloak.testsuite.JsonConfigProviderFactory not found
    at java.util.ServiceLoader.fail (ServiceLoader.java:593)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass (ServiceLoader.java:1219)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService (ServiceLoader.java:1228)
    at java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext (ServiceLoader.java:1273)
    at java.util.ServiceLoader$2.next (ServiceLoader.java:1315)
    at java.util.ServiceLoader$2.next (ServiceLoader.java:1306)
    at java.util.ServiceLoader$3.next (ServiceLoader.java:1403)
    at org.keycloak.services.resources.KeycloakApplication.loadConfig (KeycloakApplication.java:180)
    at org.keycloak.services.resources.KeycloakApplication.<init> (KeycloakApplication.java:77)
    at org.keycloak.services.resteasy.ResteasyKeycloakApplication.<init> (ResteasyKeycloakApplication.java:44)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)

```

### Anything else?

_No response_

--- a/testsuite/utils/src/main/java/org/keycloak/testsuite/JsonConfigProviderFactory.java
+++ b/testsuite/utils/src/main/java/org/keycloak/testsuite/JsonConfigProviderFactory.java
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.keycloak.services.util;
+package org.keycloak.testsuite;
 
 import org.keycloak.config.ConfigProviderFactory;
 import com.fasterxml.jackson.databind.JsonNode;
@@ -28,9 +28,10 @@ import org.jboss.logging.Logger;
 import org.keycloak.Config;
 import org.keycloak.common.util.SystemEnvProperties;
 import org.keycloak.services.ServicesLogger;
+import org.keycloak.services.util.JsonConfigProvider;
 import org.keycloak.util.JsonSerialization;
 
-public abstract class JsonConfigProviderFactory implements ConfigProviderFactory {
+public class JsonConfigProviderFactory implements ConfigProviderFactory {
 
     private static final Logger LOG = Logger.getLogger(JsonConfigProviderFactory.class);
 

